scenario/mocking.py

```diff
@@ -302,6 +303,19 @@ def secret_add(
         self._state.secrets.append(secret)
         return id
 
+    def _check_can_manage_secret(
+        self,
+        secret: "Secret",
+    ):
+        # FIXME: match real tracebacks
+        if secret.owner is None:
+            raise SecretNotFoundError("this secret is not owned by this unit/app")
```

Comment:
This is the exception when doing `self.model.get_secret(id="...")` from an application in the same model that somehow magically got hold of an ID of a different app's secret:

```
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-testing-actions-0/charm/venv/ops/model.py", line 2838, in _run
    result = subprocess.run(args, **kwargs)  # type: ignore
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '('/var/lib/juju/tools/unit-testing-actions-0/secret-get', 'secret://xxxx', '--format=json')' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-testing-actions-0/charm/./src/charm.py", line 102, in <module>
    ops.main(ActionsTestingCharm)  # type: ignore
  File "/var/lib/juju/agents/unit-testing-actions-0/charm/venv/ops/main.py", line 456, in __call__
    return main(charm_class, use_juju_for_storage=use_juju_for_storage)
  File "/var/lib/juju/agents/unit-testing-actions-0/charm/venv/ops/main.py", line 441, in main
    _emit_charm_event(charm, dispatcher.event_name)
  File "/var/lib/juju/agents/unit-testing-actions-0/charm/venv/ops/main.py", line 149, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-testing-actions-0/charm/venv/ops/framework.py", line 342, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-testing-actions-0/charm/venv/ops/framework.py", line 839, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-testing-actions-0/charm/venv/ops/framework.py", line 928, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-testing-actions-0/charm/./src/charm.py", line 44, in _on_simple_action
    self.model.get_secret(id="secret://xxxx")
  File "/var/lib/juju/agents/unit-testing-actions-0/charm/venv/ops/model.py", line 272, in get_secret
    content = self._backend.secret_get(id=id, label=label)
  File "/var/lib/juju/agents/unit-testing-actions-0/charm/venv/ops/model.py", line 3194, in secret_get
    result = self._run('secret-get', *args, return_output=True, use_json=True)
  File "/var/lib/juju/agents/unit-testing-actions-0/charm/venv/ops/model.py", line 2840, in _run
    raise ModelError(e.stderr)
ops.model.ModelError: ERROR permission denied
```